File: VB/PivotGridControl_CustomCellEdit/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraPivotGrid

Namespace PivotGridControl_CustomCellEdit

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            ' Binds the pivot grid to data.
            salesPersonTableAdapter.Fill(nwindDataSet.SalesPerson)
            ' Specifies the type of data field and format settings.
            fieldQuantity1.SummaryDisplayType = DevExpress.Data.PivotGrid.PivotSummaryDisplayType.PercentOfColumn
            fieldQuantity1.CellFormat.FormatType = DevExpress.Utils.FormatType.Custom
            fieldQuantity1.CellFormat.FormatString = "{0}%"
        End Sub

        Private Sub pivotGridControl1_CustomCellEdit(ByVal sender As Object, ByVal e As PivotCustomCellEditEventArgs)
            ' Initializes in-place editors used to represent values of regular and total cells respectively.
            Dim editorForCells As RepositoryItemProgressBar = New RepositoryItemProgressBar()
            Dim editorForTotals As RepositoryItemSpinEdit = New RepositoryItemSpinEdit()
            Dim pivot As PivotGridControl = TryCast(sender, PivotGridControl)
            pivot.RepositoryItems.AddRange(New RepositoryItem() {editorForCells, editorForTotals})
            ' Specifies editors for cells depending on a cell type.
            If e.DataField Is fieldQuantity1 Then
                If e.RowValueType = PivotGridValueType.GrandTotal Then e.RepositoryItem = editorForTotals
                If e.RowValueType = PivotGridValueType.Value Then e.RepositoryItem = editorForCells
            End If
        End Sub

        Private Sub pivotGridControl1_CustomCellValue(ByVal sender As Object, ByVal e As PivotCellValueEventArgs)
            If e.DataField Is fieldQuantity1 Then e.Value = Convert.ToDecimal(e.Value) * 100
        End Sub
    End Class
End Namespace

File: CS/PivotGridControl_CustomCellEdit/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.Data.PivotGrid;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraPivotGrid;

namespace PivotGridControl_CustomCellEdit {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        // Creates new repository items.
        RepositoryItemProgressBar riProgressBar = new RepositoryItemProgressBar();
        RepositoryItemSpinEdit riSpinEdit = new RepositoryItemSpinEdit();
        private void Form1_Load(object sender, EventArgs e) {
            // Binds the pivot grid to data.
            this.salesPersonTableAdapter.Fill(this.nwindDataSet.SalesPerson);
            // Initializes cell editors used to represent values of regular and total cells respectively.
            pivotGridControl1.RepositoryItems.AddRange(new RepositoryItem[] { riProgressBar, riSpinEdit });            
            pivotGridControl1.CustomCellEdit += new 
                EventHandler<PivotCustomCellEditEventArgs>(pivotGridControl1_CustomCellEdit);
        }
        private void pivotGridControl1_CustomCellEdit(object sender, PivotCustomCellEditEventArgs e) {
            // Specifies editors for cells depending on a cell type.
            if (e.DataField == fieldQuantityPercent) {
                if (e.RowValueType == PivotGridValueType.Value)
                    e.RepositoryItem = riProgressBar;

                if (e.RowValueType == PivotGridValueType.GrandTotal)
                    e.RepositoryItem = riSpinEdit;
            }
        }
        private void pivotGridControl1_CustomCellValue(object sender, PivotCellValueEventArgs e) {
            if (e.DataField == fieldQuantityPercent)
                e.Value = Convert.ToDecimal(e.Value) * 100;
        }
    }
}
